### DISCUSSION TITLE
Github Renovate app versus Ruby gems from private GitHub repositories

### DISCUSSION THREAD

**Discussion by springerigor** [renovatebot/renovate#15870] [Request Help]
We have enabled Renovate app for a Ruby app which is hosted in private GitHub repository. In its `Gemfile` there are dependencies hosted in other private repositories within the same GitHub organization.

```
INFO: Gemfile.lock update failed due to an unknown reason
"stderr": "fatal: could not read Username for 'https://github.com': No such device or address\nRetrying
```

What is a recommended way to guarantee that Renovate will be able to fetch the private dependencies from GitHub?

Thanks in advance 🙏

**Comment by rarkins** [comment]
How exactly are such dependencies defined? e.g. are they a git dependency, and `bundler` uses `git` under the hood to try to fetch them?

**Reply by springerigor** [reply]
```
git_source(:private_github) do |repo_name|
  "https://github.com/#{repo_name}.git"
end

gem("[GEM_NAME]", private_github: "[ORGANIZATION_NAME]/[REPO_NAME]")
```

Based on the Renovate logs, it uses `git clone` under the hood:

```
Retrying `git clone https://github.com/[ORGANIZATION_NAME]/[REPO_NAME]
```

**Comment by rarkins** [comment]
Here's the code where Renovate's artifact update gets the list of hosts/credentials to pass to bundler: https://github.com/renovatebot/renovate/blob/2e957baed962d65cb8e40136edc142af6014ad95/lib/modules/manager/bundler/artifacts.ts#L79-L81

I suspect that we don't include api.github.com in that. 

Could you create a feature request issue mirroring what's here? "Support GitHub package registry with bundler artifacts". Unfortunately if I use the "create issue from discussion" feature on github it loses all our templating.

**Reply by springerigor** [reply]
https://github.com/renovatebot/renovate/issues/15873
